flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -129,8 +133,10 @@ public boolean scaleResource(
                         scalingSummaries,
                         autoScalerEventHandler);
 
-        if (scalingWouldExceedClusterResources(
-                configOverrides.newConfigWithOverrides(conf),
+        var memoryTuningEnabled = conf.get(AutoScalerOptions.MEMORY_TUNING_ENABLED);
+        if (scalingWouldExceedMaxResources(
+                memoryTuningEnabled ? configOverrides.newConfigWithOverrides(conf) : conf,
```

Comment:
@mxm this is basically a bugfix. The original code assumed that memory tuning is always enabled which is not true.

Reply:
Did you add a test for this @gaborgsomogyi ?

Reply:
Forgotten, but now I've added.